spec/attributes_json_spec.rb

```diff
@@ -140,4 +140,24 @@ def nature
       JSON.pretty_generate(object_with_attributes_module)
     }.to raise_error(/structure too deep/)
   end
+
+  it 'converts all hash keys to string when stringify_keys: true' do
+    fixture_path = fixtures_dir + '/ZIP/arch_many_entries.zip'
+    fi_io = File.open(fixture_path, 'rb')
+
+    result = FormatParser::ZIPParser.new.call(fi_io).as_json(stringify_keys: true)
+    expect(
+      result['entries'].flat_map { |entry| entry.keys.map(&:class) }.uniq
```

Comment:
I would suggest making the expectations a bit less "smart". When dealing with tests it is usually a much more pleasant experience to be visually decoding the code under test, and not the test itself. In this instance the amount of functional sugar used occludes the fact that we want to verify that all keys we are scanning are `kind_of(String)`. Can we use a more pedestrian `each_pair` (or alike) to the same effect?

Reply:
Sure, I agree the tests should be less "smart"!

Are you suggesting something like this?

```ruby
result['entries'].each do |entry|
  entry.each do |key, value|
    expect(key).to be_a(String)
  end
end
```

I couldn't see a simpler way other than that.

Reply:
Yep that's perfect! If you opt for `#each` you might want to group the key-value using `()` though 😉

Reply:
From [what I've seen](https://stackoverflow.com/questions/46435551/in-ruby-whats-the-advantage-of-each-pair-over-each-when-iterating-through-a), `each_pair` is an alias to `each`, isn't it?

>you might want to group the key-value using () though

I didn't get it. Could you give me an example pls?

Reply:
Sure! I think `some_hash.each do |(key, value)|` is required instead of `some_hash.each do |key, value|` so that the array argument to the block is expanded into two variables. Or you can use `each_pair` instead

Reply:
cool! I didn't know about this [parenthesis rule for block variables](https://stackoverflow.com/questions/30326964/rules-on-parenthesis-for-block-variables)!

Reply:
Sorry for the long discussion about it, but I found interesting to know how it works and I think it's good to share it with you!

From what I saw, it is not necessary to use the `()` here. It would be necessary if the hash contained arrays as values and I wanted to access their values like this:

<img width="1024" alt="image" src="https://user-images.githubusercontent.com/302303/87443433-dae9a680-c5cb-11ea-82a9-bb3b76d5705c.png">

In this case, there are no arrays inside the hashes, and I just want to check their keys, so it makes no difference using `()`:

<img width="1435" alt="image" src="https://user-images.githubusercontent.com/302303/87444045-a7f3e280-c5cc-11ea-803a-8a134e4e44e7.png">

And also makes no difference using `each` or `each_with_pair` given they are aliases!

<img width="1423" alt="image" src="https://user-images.githubusercontent.com/302303/87444326-0a4ce300-c5cd-11ea-9070-6ce27e676500.png">